api/all.yaml

```diff
@@ -1316,6 +1322,22 @@ components:
         external_language:
           description: External language of the function.
           type: string
+      required:
+        - name
+        - catalog_name
+        - schema_name
+        - input_params
+        - data_type
+        - full_data_type
+        - routine_body
+        - routine_definition
+        - parameter_style
+        - is_deterministic
+        - sql_data_access
+        - is_null_call
+        - security_type
+        - specific_name
+        - properties
```

Comment:
I believe many fields here (like properties) should be optional. Is it the case that in the current API implementation, they're required?

Reply:
I don't think they are required

Reply:
Thank you for the comment. These lines are just reformatting the previous implementation (specifically, moving the required field to the end of the CreateFunction object). I will update this implementation to align with the current API specification. Could you please provide a reference to the current API specification or guide which fields to be optional?

Reply:
Hi Shingo, I think we should just remove properties from required for now, based on the implementation, others are required. thanks!

Reply:
Thanks for the guide, Vikrant! I've updated the PR to make the `properties` field optional in the `CreateFunction` and `FunctionInfo` objects.

cc: @haogang @vikrantpuppala